lifecycle-management/Ahk.Lifecycle.Management/Data/CosmosDbRepository.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Ahk.Lifecycle.Management
+{
+    public class CosmosDbRepository : IRepository
+    {
+        private readonly CosmosDbContext _context;
+        private volatile bool created;
+        public CosmosDbRepository(CosmosDbContext context) => _context = context;
+
+        private async Task createDB()
+        {
+            if(!created)
+            {
+                created = true;
+                await _context.Database.EnsureCreatedAsync();
+            }
+        }
+
+        public async Task Insert(LifecycleEvent data)
+        {
+            await this.createDB();
+
+            var newStatistics = new RepositoryStatistics()
+            {
+                Repository = data.Repository,
+                Username = data.Username,
+                EventCount = 1,
+                LastEventType = data.EventType,
+                LastEventDate = data.Timestamp,
+                LastEventId = data.LifecycleEventId
+            };
+
+            await _context.LifecycleEvents.AddAsync(data);
+            await _context.SaveChangesAsync();
+
+            await this.Insert(newStatistics);
+        }
+
+        public async Task<ICollection> GetOne(string repo)
+        {
+            await this.createDB();
+
+            return await Task.FromResult<ICollection>(_context.LifecycleEvents.Where(e => e.Repository.Contains(repo)).ToList());
+        }
+        public async Task Insert(RepositoryStatistics data)
+        {
+            await this.createDB();
+
+            try
+            {
+                await _context.Statistics.AddAsync(data);
+                await _context.SaveChangesAsync();
+            }
+            catch (Microsoft.EntityFrameworkCore.DbUpdateException ex)
+            {
+                Console.WriteLine(ex.Message);
+                await this.Update(data);
+            }
+        }
+
+        public async Task Update(RepositoryStatistics data)
+        {
+            await this.createDB();
+
+            _context.Statistics.Remove(data);
+            var oldStat = _context.Statistics.FirstOrDefault(o => o.Repository == data.Repository);
+
+            if (oldStat != null)
+            {
+                if (oldStat.LastEventDate < data.LastEventDate)
+                {
+                    oldStat.LastEventDate = data.LastEventDate;
+                    oldStat.LastEventId = data.LastEventId;
+                    oldStat.LastEventType = data.LastEventType;
+                }
+                
+                oldStat.EventCount++;
+
+                _context.Statistics.Update(oldStat);
+                await _context.SaveChangesAsync(true);
+            }
+        }
+
+        public async Task<ICollection> GetMany(string repository = "", string username = "", int page = 1, int limit = 10)
```

Comment:
ez elég furán néz ki, mi a visszatérési értéke? miért nem típusos?

Reply:
Jelenleg ez a függvény végzi el a keresést a RepositoryStatistics-ok között és állítja össze a végeredményt. Azért nem típusos, mert nem debug szempontjából nagyobb teret enged meg visszatérések terén (pl. Events-ek közül is tudok visszaadni értéket).
Végeredményben egy pl. SearchResult osztály lenne a visszatérési értéke, ami

```
{
  count: int,
  page: int,
  limit: int,
  results: RepositoryStatistics[]
}
```

formájú osztály lenne.